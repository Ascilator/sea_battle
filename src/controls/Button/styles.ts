import styled, { css } from "styled-components";
import { StylesType } from "@/types";

export const StyledButton = styled.button<StylesType>`
  position: relative;
  user-select: none;
  cursor: pointer;
  color: #eee;
  text-transform: uppercase;
  font-weight: 700;
  letter-spacing: 0.05em;

  & span {
    position: relative;
    z-index: 1;
    display: block;
    padding: 10px;
    border-radius: 1000px;
    background-color: #d99d61;
    border: 1px solid darken(#d99d61, 5%);
    box-shadow: -5px 5px 10px rgba(0, 0, 0, 0.2);
    transition: transform 0.1s, box-shadow 0.2s, background-color 0.2s;
  }
  &:after {
    content: "";
    position: absolute;
    top: -10px;
    bottom: -10px;
    left: -10px;
    right: -10px;
    border-radius: 1000px;
    box-shadow: 0 0 0px rgba(0, 0, 0, 0.2), inset 0 0 0px rgba(0, 0, 0, 0.2);
    transition: box-shadow 0.1s;
  }

  &:hover,
  &:focus {
    & span {
      background-color: lighten(#d99d61, 3%);
    }
  }
  &:active {
    & span {
      background-color: #d99d61;
      transform: scale(0.97);
      box-shadow: -2px 2px 5px rgba(0, 0, 0, 0.2);
    }
    &:after {
      box-shadow: 0 0 2px rgba(0, 0, 0, 0.2),
        inset -3px 3px 1em rgba(0, 0, 0, 0.2);
    }
  }
`;
