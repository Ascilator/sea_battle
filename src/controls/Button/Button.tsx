import { FC } from 'react';
import { ButtonProps } from '@/types';

import { StyledButton } from './styles';

const Button: FC<ButtonProps> = ({ text, color, onClick }) => (
  <StyledButton color={color} onClick={onClick}>
    <span>{text}</span>
  </StyledButton>
);

export { Button };
