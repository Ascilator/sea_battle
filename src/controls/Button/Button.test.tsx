import renderer from 'react-test-renderer';
import { Button } from './Button';

test('Button render', () => {
  const buttonGreen = renderer.create(<Button text="test btn" color="#0f0" />).toJSON();
  expect(buttonGreen).toMatchSnapshot();

  const buttonBlue = renderer.create(<Button text="test btn" color="#0ff" />).toJSON();
  expect(buttonBlue).toMatchSnapshot();

  const buttonText = renderer.create(<Button text="button with text" color="#0f0" />).toJSON();
  expect(buttonText).toMatchSnapshot();
});
