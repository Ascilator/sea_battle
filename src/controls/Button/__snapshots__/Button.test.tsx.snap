// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render 1`] = `
<button
  className="sc-bdvvtL deOBat"
  color="#0f0"
  onClick={[Function]}
>
  <span>
    test btn
  </span>
</button>
`;

exports[`Button render 2`] = `
<button
  className="sc-bdvvtL jlnfRj"
  color="#0ff"
  onClick={[Function]}
>
  <span>
    test btn
  </span>
</button>
`;

exports[`Button render 3`] = `
<button
  className="sc-bdvvtL deOBat"
  color="#0f0"
  onClick={[Function]}
>
  <span>
    button with text
  </span>
</button>
`;
